feat(campfire): add template for Once Campfire

elalemanyo opened the pull request:
## Changes
- Added [once-campfire](https://github.com/basecamp/once-campfire) app template.

elalemanyo committed:
feat(campfire): add Docker Compose configuration for Once Campfire service

elalemanyo committed:
fix(campfire): update port configuration from 80 to 3000 in Docker Compose file

elalemanyo committed:
style(campfire): format environment variables for better readability in Docker Compose file

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+services:
+  campfire:
+    build: https://github.com/basecamp/once-campfire.git#${BRANCH:-main}
+    ports:
+      - "3000:80"
```
Why is the need for port mapping here?

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+
+services:
+  campfire:
+    build: https://github.com/basecamp/once-campfire.git#${BRANCH:-main}
```
All the one click services we have on Coolify uses prebuilt docker images, we don't build the entire service on the user's server.

elalemanyo replied:
@ShadowArcanist  Thanks for the review! You’re right — one-clicks should use prebuilt images. Campfire upstream is MIT-licensed and documents Docker deployment but doesn’t publish an image yet. I can:

1.	Switch the template to an official ghcr.io/basecamp/once-campfire image as soon as they publish (there’s an upstream PR to add GHCR), or
2.	Build a community image under ghcr.io/coollabsio/once-campfire (multi-arch) and point the template to that.

Let me know which route you prefer

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+    volumes:
+      - campfire-storage:/rails/storage
+    environment:
+      SECRET_KEY_BASE: ${SECRET_KEY_BASE}     # required, set this to a secure random value
+      VAPID_PUBLIC_KEY: ${VAPID_PUBLIC_KEY}   # optional, for notifications
+      VAPID_PRIVATE_KEY: ${VAPID_PRIVATE_KEY} # optional, for notifications
+      DISABLE_SSL: ${DISABLE_SSL-true}        # uncomment to disable SSL
+      SSL_DOMAIN: ${SSL_DOMAIN-false}         # optional, for SSL
+      SKIP_TELEMETRY: ${SKIP_TELEMETRY-true}  # optional, set to disable telemetry
+      SENTRY_DSN: ${SENTRY_DSN}               # optional, for error reporting
```
Please check the compose files we have on https://github.com/coollabsio/coolify/tree/v4.x/templates/compose and follow same styling of env by formatting them like for example: `SKIP_TELEMETRY=${SKIP_TELEMETRY:-true}`

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+    volumes:
+      - campfire-storage:/rails/storage
+    environment:
+      SECRET_KEY_BASE: ${SECRET_KEY_BASE}     # required, set this to a secure random value
```
Use Coolify's magic env for random value generate like `SERVICE_PASSWORD_CAMPIRE`

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+# category: messaging
+# tags: campfire,chat,communication,rails,once,basecamp,37signals
+# logo: svgs/once-campfire.png
+# port: 3000
```
Application port is 80 (based on the campfire port mapping seen below)

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+    restart: unless-stopped
+    volumes:
+      - campfire-storage:/rails/storage
+    environment:
```
`SERVICE_URL_CAMPFIRE` is missing..

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+      SECRET_KEY_BASE: ${SECRET_KEY_BASE}     # required, set this to a secure random value
+      VAPID_PUBLIC_KEY: ${VAPID_PUBLIC_KEY}   # optional, for notifications
+      VAPID_PRIVATE_KEY: ${VAPID_PRIVATE_KEY} # optional, for notifications
+      DISABLE_SSL: ${DISABLE_SSL-true}        # uncomment to disable SSL
```
The comment says `# uncomment to disable SSL` but the env is not commented out at all, also user can just set the value to false no?

ShadowArcanist commented on templates/compose/once-campfire.yaml:
```diff
@@ -0,0 +1,31 @@
+      timeout: 5s
+      retries: 5
+
+volumes:
+  campfire-storage:
```
Coolify adds these automatically, no need to manually add these

ShadowArcanist commented:
Please check the other templates on https://github.com/coollabsio/coolify/tree/v4.x/templates/compose to get an idea on how envs, docker images were used on service templates.

Also make sure to test the template before sending the PR

elalemanyo committed:
fix(campfire): correct port comment from 3000 to 80 in Docker Compose file

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

elalemanyo committed:
refactor(campfire): streamline environment variable definitions in Docker Compose file

elalemanyo committed:
style(campfire): update comment for DISABLE_SSL environment variable for clarity

elalemanyo commented:
@ShadowArcanist Thanks for pointing that out! 🙈 I went through the other templates and updated mine to match their style (envs, image usage, removed extra volumes, etc.).
I also tested the template locally before opening the PR by creating a new service in Coolify and pasting the docker-compose content — it started up and worked as expected.
Let me know if you’d like me to test it in a different way, but this seems to be how the other one-click templates are validated too.

FYI: @Cinzya

elalemanyo committed:
fix(campfire): update service definition to use image instead of build in Docker Compose file

ShadowArcanist reviewed (commented):
LGTM

ShadowArcanist commented:
Hey @elalemanyo, please submit a PR to the docs repository as suggested by the GitHub Actions bot above. Once that’s done, let me know here — the 'waiting for changes' label has been added for this.

elalemanyo commented:
@ShadowArcanist done https://github.com/coollabsio/coolify-docs/pull/389 🙂

andrasbacsai commented:
Thank you for the PR!
